apiserver/facades/client/bundle/bundle.go

```diff
@@ -248,7 +248,44 @@ func (b *BundleAPI) fillBundleData(model description.Model) (*charm.BundleData,
 			Annotations: machine.Annotations(),
 			Series:      machine.Series(),
 		}
-		if result := b.constraints(machine.Constraints()); len(result) != 0 {
+
+		result := b.constraints(machine.Constraints())
+
+		// Get the machine, so we can get the hardware characteristics.
+		// This is done for parity with GUI.
+		if machineState, _ := b.backend.Machine(machine.Id()); machineState != nil {
```

Comment:
Don't ignore the error. The check should be for err != nil rather than machineState != nil